server/game/cards/events/07/scalingthewall.js

```diff
@@ -0,0 +1,50 @@
+const DrawCard = require('../../../drawcard.js');
+
+class ScalingTheWall extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            when: {
+                afterChallenge: (event, challenge) => (
+                    challenge.winner === this.controller &&
+                    challenge.attackingPlayer === this.controller &&
+                    this.hasAttackingWildling()
+                )
+            },
+            handler: () => {
+                this.game.promptForSelect(this.controller, {
+                    cardCondition: card => this.cardCondition(card),
+                    activePromptTitle: 'Select location',
```

Comment:
Instead of a manual `promptForSelect` here, it should be possible to use the `target` property. E.g.
```javascript
this.reaction({
    when {
        // ...
    },
    target: {
        cardCondition: card => this.cardCondition(card),
        activePromptTitle: 'Select location'
    },
    handler: context => {
        // the chosen card will be stored on context.target
    }
});
```

See [documentation](https://github.com/cryogen/throneteki/blob/master/docs/implementing-cards.md#choosing--targeting-cards) for more details (the examples given are for actions, but it works the same with interrupts and reactions)